tests/components/mqtt/conftest.py

```diff
@@ -121,3 +121,148 @@ def record_calls(msg: ReceiveMessage) -> None:
         recorded_calls.append(msg)
 
     return record_calls
+
+
+@pytest.fixture(name="supervisor")
+def supervisor_fixture() -> Generator[MagicMock]:
+    """Mock Supervisor."""
+    with patch(
+        "homeassistant.components.mqtt.config_flow.is_hassio", return_value=True
+    ) as is_hassio:
+        yield is_hassio
+
+
+@pytest.fixture(name="discovery_info")
+def discovery_info_fixture() -> Any:
+    """Return the discovery info from the supervisor."""
+    return DEFAULT
+
+
+@pytest.fixture(name="get_addon_discovery_info", autouse=True)
+def get_addon_discovery_info_fixture(discovery_info: Any) -> Generator[AsyncMock]:
+    """Mock get add-on discovery info."""
+    with patch(
+        "homeassistant.components.hassio.addon_manager.async_get_addon_discovery_info",
+        return_value=discovery_info,
+    ) as get_addon_discovery_info:
+        yield get_addon_discovery_info
+
+
+@pytest.fixture(name="addon_setup_time", autouse=True)
+def addon_setup_time_fixture() -> Generator[int]:
+    """Mock add-on setup sleep time."""
+    with patch(
+        "homeassistant.components.mqtt.config_flow.ADDON_SETUP_TIMEOUT", new=0
+    ) as addon_setup_time:
+        yield addon_setup_time
+
+
+@pytest.fixture(name="addon_store_info")
+def addon_store_info_fixture() -> Generator[AsyncMock]:
+    """Mock Supervisor add-on store info."""
+    with patch(
+        "homeassistant.components.hassio.addon_manager.async_get_addon_store_info"
+    ) as addon_store_info:
+        addon_store_info.return_value = {
+            "available": False,
+            "installed": None,
+            "state": None,
+            "version": "1.0.0",
+        }
+        yield addon_store_info
+
+
+@pytest.fixture(name="addon_info")
+def addon_info_fixture() -> Generator[AsyncMock]:
```

Comment:
Let's not create a 7th copy of these fixtures.
Instead:
- The existing fixture implementations should be refactored to helper functions which are moved from to `tests/components/hassio/conftest.py` to `tests/components/hassio/common.py`
- Fixtures added to `tests/components/conftest.py`

In this PR, hassio and mqtt should make use of the shared fixtures provided by `hassio`
In follow-up PRs, homeassistant_hardware, homeassistant_sky_connect, homeassistant_yellow, matter and otbr should be updated to use the shared fixtures.

Reply:
Right so for this PR, is it okay to add the new hassio fixtures to `tests/components/conftest.py`?

Reply:
I moved the shared fixtures to `tests/components/conftest.py`. In a follow up PR the redundant fixtures can be cleanup, and refactored if needed. In most cases they are copied now.